doc/contrib/recording.md

```diff
@@ -0,0 +1,82 @@
+# Session Recording Technical Documentation
+
+The purpose of this document is to explain in detail the concepts and implementation of Elektra's session recording feature.
+While the feature is under development, it acts as sort of in-between documentation, roadmap and TODO list.
+
+## What is Session Recording?
+
+After every `kdbSet` action, the changes are calculated using Elektra's powerful changetracking API.
+The result of the calculation is an `ElektraDiff` instance we'll call _part diff_ throughout this document.
+The session recording plugin merges those _part diffs_ together and creates and persists an overall _session diff_.
+Importantly, the _session diff_ is persisted cross-process.
```

Comment:
That's true. But then again that is the case for all of Elektra. So I don't see a good way to handle errors here. I mean we could use a system-wide semaphore/lock in `kdbSet` when record is enabled ...

Reply:
Many conflicts can be solved automatically. Because the conflict detection is on a file-level in `resolver`, most of the time there isn't actually a conflict on a per-key level. That's why the `merge` library exists. Some special cases can even be solved with a simple loop. See e.g., `libelektra-highlevel` (*).

However, my main point here was that the current documentation makes it sound like session recording is somehow  (more) resistant to inter-process conflicts. Which is feasible, if you don't know the internals, since the session could be stored in an inherently conflict-free append-only fashion (e.g. store each `part diff` in a unique file and only merge into a `session diff` when needed).

(*) For the unlikely conflict that the exact key we have in `elektraSaveKey` was modified, it's just "last write wins" for that key.

Reply:
How about:

```suggestion
Importantly, the _session diff_ is shared across processes.
The _session diff_ persistently records all changes made to the whole KDB from any process between the start and end of the session.
Because of the cross-process nature of a _session diff_, it is also susceptible to "simultaneous write" conflicts.
We tried to limit the impact of these conflicts through locking.
```

I think that explains the whole situation a bit better.

Regarding the "file-based locking": As you can see from my review, I'm not the biggest fan. But it is certainly better than before (see latest edit of review).